src/actinia_core/core/stac_importer_interface.py

```diff
@@ -0,0 +1,214 @@
+    has_plugin = False
+
+
+class STACImporter:
+
+    @classmethod
+    def stac_import(self, stac_collecition_id, semantic_label, interval, bbox, filter):
+
+        if has_plugin:
+            try:
+                stac_name = stac_collecition_id.split(".")[3]
+            except Exception:
+                raise AsyncProcessError("The source has not the right structure")
+
+            stac_root = self.get_search_root(stac_collecition_id)
+
+            stac_filtered = self.apply_filter(stac_root, stac_name,
+                                              interval, bbox, filter)
+
+            stac_result = self.get_filtered_bands(stac_filtered, semantic_label)
+
+            stac_processes = []
+
+            for key, value in stac_result.items():
+
+                for name_id, url in value.items():
+
+                    output_name = stac_name + "_" + key + "_" + name_id
+
+                    # From Here Onwards, the Process build starts
+                    exec_params = ["input=%s" % "/vsicurl/"+url,
+                                   "output=%s" % output_name,
+                                   "-o"]
+
+                    p = Process(
+                        exec_type="grass",
+                        executable="r.in.gdal",
+                        executable_params=exec_params,
+                        id=f"r_gdal_{os.path.basename(output_name)}",
+                        skip_permission_check=True
+                    )
+
+                    stac_processes.append(p)
+        else:
+            raise AsyncProcessError("Actinia STAC plugin is not installed")
+
+        return stac_processes
+
+    @classmethod
+    def get_search_root(self, stac_collecition_id):
+
+        stac_from_actinia = callStacCollection(stac_collecition_id)
+
+        stac_json = json.loads(stac_from_actinia)
+
+        for item in stac_json["links"]:
+            if item["rel"] == "root":
+                stac_url = item["href"]
+
+                if "?" in stac_url:
+                    stac_url = stac_url.split("?")[0]
+
+                if stac_url.endswith("/"):
+                    stac_root_search = stac_url + "search"
+                else:
+                    stac_root_search = stac_url + "/search"
+
+        return stac_root_search
+
+    @classmethod
+    def apply_filter(self, stac_root_search, stac_name, interval, bbox, filter):
+
+        search_body = {
+            "collections": [stac_name],
+        }
+        search_body["query"] = filter
+
+        search_body["bbox"] = bbox
+
+        search_body["interval"] = interval
+
+        stac_search = requests.post(
+            stac_root_search,
+            json=search_body
+        )
+
+        full_filtered_result = stac_search.json()
+
+        if "features" in full_filtered_result:
+            return full_filtered_result
+        else:
+            raise AsyncProcessError(full_filtered_result)
+
+    @classmethod
+    def get_filtered_bands(self, stac_items, semantic_label):
```

Comment:
Is a classmethod needed here? Or would it make more sense to use the indication that this method is only used inside this class (`_get_filtered_bands` style). - Same comment goes to all methods here except `get_stac_import_download_commands`. And to my knowledge it is a convention that `cls` is used instead of `self` when adressing this method, but I am no expert on this.

Maybe you can indicate why you used a classmethod here and we can discuss what makes more sense!?

Reply:
I couldn't handle `@classmethod ` and `self` functions to work together. So far what I understood is that you can not call a self function inside a `@classmethod `, one should either `@staticmethods ` and  `self` functions  for our purpose and the class method should be focus on object characteristics. (somehow it makes sense to me)

Reply:
Ok, makes sense.